SQL/Archive/2017-11-01-use_consent_to_database.sql

```diff
@@ -0,0 +1,35 @@
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'useConsent', 'Enable if the study uses the loris architecture for consent', 1, 0, 'boolean', ID, 'Use consent', 15 FROM ConfigSettings WHERE Name='study';
+INSERT INTO Config (ConfigID, Value) SELECT ID, 'false' FROM ConfigSettings WHERE Name='useConsent';
+
+CREATE TABLE `consent_type` (
+  `ConsentTypeID` int(2) NOT NULL AUTO_INCREMENT,
+  `Name` varchar(255) NOT NULL,
+  `Label` varchar(255) NOT NULL,
+  CONSTRAINT `PK_consent_type` PRIMARY KEY (`ConsentTypeID`),
+  CONSTRAINT `UK_consent_type_Name` UNIQUE KEY `Name` (`Name`),
+  CONSTRAINT `UK_consent_type_Label` UNIQUE KEY `Label` (`Label`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+CREATE TABLE `candidate_consent_type_rel` (
+  `CandidateID` int(6) NOT NULL,
+  `ConsentTypeID` int(2) NOT NULL,
+  `Status` enum('yes', 'no', 'not_answered') NOT NULL,
+  `DateGiven` date DEFAULT NULL,
+  `DateWithdrawn` date DEFAULT NULL,
+  CONSTRAINT `PK_candidate_consent_type_rel` PRIMARY KEY (`CandidateID`,`ConsentTypeID`),
+  CONSTRAINT `FK_candidate_consent_type_rel_CandidateID` FOREIGN KEY (`CandidateID`) REFERENCES `candidate` (`CandID`),
+  CONSTRAINT `FK_candidate_consent_type_rel_ConsentTypeID` FOREIGN KEY (`ConsentTypeID`) REFERENCES `consent_type` (`ConsentTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+CREATE TABLE `candidate_consent_type_history` (
+  `CandidateConsentHistoryID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `PSCID` varchar(255) NOT NULL,
+  `ConsentName` varchar(255) NOT NULL,
+  `ConsentLabel` varchar(255) NOT NULL,
+  `Status` enum('yes','no','not_answered') NOT NULL,
+  `DateGiven` date DEFAULT NULL,
+  `DateWithdrawn` date DEFAULT NULL,
+  `EntryStaff` varchar(255) DEFAULT NULL,
+  `EntryDate` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  CONSTRAINT `PK_candidate_consent_type_history` PRIMARY KEY (`CandidateConsentHistoryID`)
```

Comment:
varchar() should be after date and timestamp field.

Reply:
could you explain why the the varchar comes after the date and timestamp?

Reply:
@zaliqarosli For efficiency, if DB engine want to read a field placed after, it as to traverse the varchar() until the end. Can't jump directly to the required field start as the offset from row start in not fixed.
I'm preparing a presentation on that for an ADM.

Reply:
ahh okay i think i know what you mean. looking forward to the adm!